### ISSUE TITLE
Assigning existing Organization API key to new/updated projects

### ISSUE THREAD

**Issue by jmccarty3** [mongodb/terraform-provider-mongodbatlas#2030]
It looks like with the release of provider version 1.12.0 the ability to associate existing Organization keys with a project has been lost in the provider.

Looking at issues such as:
- https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1369
- https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1278

It looks like the assumption now is that some project has to own an API key and then multiple projects in an organization can be assigned roles. That breaks top-down control flows where some organization keys are created centrally and shared down to independent projects for usage. An example would be keys for HashiCorp Vault or other in-house automation workflows.

It looks like this feature is still supported via the direct API with https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Programmatic-API-Keys/operation/addProjectApiKey

Is there a missing resource that could be added along the lines of `mongodbatlas_project_api_key_assignment` that takes an existing key ID and makes a non-exclusive assignment to the project?

**Comment by AgustinBettati** [comment]
Hello @jmccarty3.

The most significant changes related to the handling of PAKs came in version 1.10.0, where [docs/guides/Programmatic-API-Key-upgrade-guide-1.10.0](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/Programmatic-API-Key-upgrade-guide-1.10.0) has details of the changes that where made and its motivations. 

With that said [mongodbatlas_project_api_key](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project_api_key) is the resource that handles the assignments of projects with specific project-level roles. This resource does in fact handle the creation of the api key as well. 

To confirm my understanding, you would like to leverage existing org level api keys (potentially defined with [mongodbatlas_api_key](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/api_key)) with the possibility of managing assignments of projects for that key.

One alternative is to import this key into the [mongodbatlas_project_api_key](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project_api_key) resource and then define or remove project assignments accordingly. This does have the limitation that the key needs at least one project assignment in order to be able to import the resource.

**Comment by jmccarty3** [comment]
@AgustinBettati Yes I have read through the documentation around the change to using PAKs with projects from the upgrade. For many scenarios I also agree with the motivations for creating/working with the keys in this new manner.

However, your understanding of my ask is correct. I would like to assign existing Organization keys to projects and manage project assignment outside of the new PAK model. Using a "shadow" project to own an organization key that is used across multiple projects does not really help for two main reasons.

1. It creates "cruft" projects for the sole purpose of "owning" a key and not actually being useful. If there is no other need for the project, why create it to only own keys?

2. It still creates an issue where a central object must know about all projects that require using the key which introduces both management overhead and requires more coordination between groups in a distributed organization.

Since the API still explicitly supports adding a key to a project, does a new resource type possibly make sense? Or is that API feature possibly slated to be removed?

**Comment by AgustinBettati** [comment]
We currently have the following internal ticket [CLOUDP-215112](https://jira.mongodb.org/browse/CLOUDP-215112) to track work related to this limitation. We will provide updated on timelines once we have more details.

**Comment by aristosvo** [comment]
Is it possible to get an update on this issue?

One reason we need this is the need to assign only projects, not have the apiKey data itself in state.

We basically would like to have the equivalent of `atlas projects apiKeys assign <ID> [options]`.

I created [a PR](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3198) containing an example of the resource I think we need, just didn't docs yet. I would happily work together with you on fixing this forward. I've contributed to other Terraform providers before, so the technical part shouldn't be an issue.

Thanks a lot!

**Comment by marcosuma** [comment]
Thank you @aristosvo, I really appreciated your patience. I've left a comment in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3198 directly.

**Comment by aristosvo** [comment]
@marcosuma Can you give a heads-up on the timelines here? A bit of openness would help a lot here.

Does it help if I create a ticket requesting this specific feature as well?

**Comment by oarbusi** [comment]
Hi @aristosvo
This has been released in v1.38.0 of the provider. Check out the [Upgrade guide ](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.38.0-upgrade-guide)for more details
